### PR TITLE
Updating scapegoat sbt plugin

### PR DESCRIPTION
**Pull request by BalmungSan**
Hello @mwz ,

I had updated the scapegoat sbt plugin and the scapegoat version to their most recent releases.
I also configure the scapegoat plugin to only generate the xml report, hope you don't need the others.

### CODE REVIEW

**Comment by mwz** [comment]
I've just seen the new release, thanks for upgrading!
Yes, we'll need only the xml report, same thing goes for scoverage, but I haven't checked yet whether you can exclude the other reports.

Btw, I'm setting up a public instance of SonarQube, so we can actually run the analysis against this plugin, including PRs and releases.

**Comment by BalmungSan** [comment]
Oh, configuring the scocerage report is easy, its just a setting in the build.sbt file, if you want I will commit to this PR the change in a hour or so.

And the public sonarqube instance for project sounds like a great idea.

Obtener Outlook para Android<https://aka.ms/ghei36>



De: Michael Wizner
Enviado: lunes, 19 de febrero 15:59
Asunto: Re: [mwz/sonar-scala] Updating scapegoat sbt plugin (#19)
Para: mwz/sonar-scala
Cc: Luis Miguel Mejía Suárez, Author


I've just seen the new release, thanks for upgrading!
Yes, we'll need only the xml report, same thing goes for scoverage, but I haven't checked yet whether you can exclude the other reports.
Btw, I'm setting up a public instance of SonarQube, so we can actually run the analysis against this plugin, including PRs and releases.

File: build.sbt [hunk 1]
```diff
@@ -20,12 +20,16 @@ scalacOptions := Seq(
   "-feature",
   "-language:reflectiveCalls"
 )
-scalacOptions in Scapegoat += "-P:scapegoat:overrideLevels:TraversableHead=Warning:OptionGet=Warning"
+scalacOptions in Scapegoat += "-P:scapegoat:overrideLevels:TraversableHead=Warning:OptionGet=Warning:TraversableLast=Warning"
```
Comment (mwz) [inline build.sbt]: The goal is to address all of the errors flagged by Scapegoat, but it's ok for now - I'll raise another issue to address that separately.

File: build.sbt [hunk 2]
```diff
@@ -20,12 +20,16 @@ scalacOptions := Seq(
...
 javacOptions := Seq("-Xlint:deprecation")
 cancelable in Global := true
 scalafmtOnCompile in ThisBuild := true
-scapegoatVersion in ThisBuild := "1.3.3"
+scapegoatVersion in ThisBuild := "1.3.4"
 scapegoatConsoleOutput := false
+scapegoatReports := Seq("xml")
+coverageOutputXML := true
+coverageOutputHTML := false
+coverageOutputCobertura := false
```
Comment (mwz) [inline build.sbt]: 👍

Comment (mwz) [reply build.sbt]: Thanks for adding those settings.

File: build.sbt (hunk 1 above)
Comment (BalmungSan) [reply build.sbt]: yeah of course, if not what would be the case of having scapegoat in a first place.
I would be happy to help removing some of those errors, I will raise a PR whenever I remove all I could.

**Pull request merged**